wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift

```diff
@@ -113,32 +131,60 @@ public final class ConversationService: ConversationServiceInterface {
                 allowServices: allowServices
             ),
             enableReceipts: enableReceipts,
-            messageProtocol: messageProtocol,
-            completion: completion
-        )
+            messageProtocol: messageProtocol) { result in
+                switch result {
+                case .failure(.networkError(.unreachableDomains(let domains))):
+                    retry(excludingDomains: domains)
+
+                default:
+                    completion(result)
+                }
+            }
     }
 
     private func internalCreateGroupConversation(
         name: String?,
         users: Set<ZMUser>,
         completion: @escaping (Swift.Result<ZMConversation, ConversationCreationFailure>) -> Void
     ) {
+        func retry(excludingDomains domains: Set<String>) {
+            let usersToExclude = users.belongingTo(domains: domains)
+            let usersToAdd = Set(users).subtracting(usersToExclude)
+            internalCreateGroupConversation(
+                teamID: nil,
+                name: name,
+                users: usersToAdd,
+                failedToAddUsers: usersToExclude,
+                accessMode: ConversationAccessMode(),
+                accessRoles: [],
+                enableReceipts: false,
+                messageProtocol: .proteus,
+                completion: completion)
+        }
```

Comment:
would it make sense to have this as a separate method to avoid duplication ?

Reply:
yes, it's popssible